src/capi/typeobject.cpp

```diff
@@ -623,6 +625,65 @@ static PyObject* slot_tp_iternext(PyObject* self) noexcept {
     return call_method(self, "next", &next_str, "()");
 }
 
+static PyObject* slot_tp_getattro(PyObject* self, PyObject* name) {
+    static PyObject* getattribute_str = NULL;
+    return call_method(self, "__getattribute__", &getattribute_str, "(O)", name);
+}
+
+static PyObject* call_attribute(PyObject* self, PyObject* attr, PyObject* name) {
+    PyObject* res, * descr = NULL;
+    descrgetfunc f = Py_TYPE(attr)->tp_descr_get;
+
+    if (f != NULL) {
+        descr = f(attr, self, (PyObject*)(Py_TYPE(self)));
+        if (descr == NULL)
+            return NULL;
+        else
+            attr = descr;
+    }
+    res = runtimeCall(attr, ArgPassSpec(1, 0, true, true), name, NULL, NULL, NULL, NULL);
```

Comment:
I think this function throws both C++ and CAPI-style exceptions (it propagates exceptions from the C++ `runtimeCall` and the CAPI `f`) -- we should pick one type of exception for it to throw, and have it convert the other one.  I'd bias towards CAPI-style since this looks like a CAPI function (uses CPython's naming conventions, type names, etc).

To make it explicit which style it is, I've been trying to add `noexcept` specifiers to any CAPI functions.  Can we add those in this PR as appropriate?